GetIntoTeachingApi/Models/Crm/BaseModel.cs

```diff
@@ -148,6 +150,11 @@ private static string TrimAndNullifyIfEmpty(string input)
             return string.IsNullOrWhiteSpace(input) ? null : input;
         }
 
+        private bool HasGeneratedUpfrontId()
+        {
+            return this is ICanGenerateUpfrontId && ((ICanGenerateUpfrontId)this).ApiOnlyId;
```

Comment:
I think we should have `BaseModel` implement `ICanGenerateUpfrontId` instead of `Candidate` - it would make sense to have `HasGeneratedUpfrontId` on the interface as well